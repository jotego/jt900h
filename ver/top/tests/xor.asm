    ; CPL
    main section code
    org 0
    ld a,0xbf    ; common header

    ld a,0xf0
    ld b,0x0f
    xor a,b
    cp a,0xff
    jp ne,bad_end

    or c,1
    ld wa,(data)
    xor wa,0x3501
    cp wa,0xffff
    jp ne,bad_end

    or c,2
    ld xhl,0xff0000
    xor xhl,0x800000
    cp xhl,0x7f0000
    jp ne,bad_end

    or c,4

end_loop:
    ld hl,0xbabe
    ld (0xffff),0xff
    jp end_loop
bad_end:
    ld hl,0xdead
    ld (0xffff),0xff
    jp bad_end
data:
    dw 0xcafe,0xbeef,0xffff,0xeeee,0xcccc
    end