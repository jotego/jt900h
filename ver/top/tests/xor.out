0000ffff - 00000000 - 00000000 - 00000000
00000f07 - 00000000 - 00000000 - 00000000
00000000 - 00000000 - 00000000 - 00000000
007fbabe - 00000000 - 00000000 - 00000000
-- index registers --
00000000 - 00000000 - 00000000 - 00000000
